### PR TITLE
HDDS-8652. [Snapshot] Added DB sequence number to the compaction log entry

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
This change is to add DB sequence number to the compaction log entry so that it can be used to ignore any newer compaction log entries than the check-pointed active DB's sequence number.
more detail: https://github.com/apache/ozone/pull/4680#issuecomment-1553565271

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8652

## How was this patch tested?
Updated existing unit tests.

### CODE REVIEW

File: hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java [hunk 1]
```diff
@@ -282,22 +282,22 @@ public void testGetSSTDiffListWithoutDB(String description,
...
-        + "C 000093,000090,000083:\n"
+        + "C 7 000093,000090,000083:\n"
         // Snapshot 2
         + "S 14980 e7ad72f8-52df-4430-93f6-0ee91d4a47fd " + createdTime + "\n"
-        + "C 000098,000096,000085,000078,000071,000064,000060,000052:000099\n"
-        + "C 000105,000095,000088:000107\n"
+        + "C 8 000098,000096,000085,000078,000071,000064,000060,000052:000099\n"
+        + "C 9 000105,000095,000088:000107\n"
         // Snapshot 3
         + "S 17975 4f084f6e-ed3d-4780-8362-f832303309ea " + createdTime + "\n";
```
Comment (smengcl) [inline hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java:302]: Should be in monotonically increasing order considering the "S " entries? e.g.

```
S 1000
C 2000
S 3008
...
```

**Review by smengcl — APPROVED** [review]
lgtm

**Review by prashantpogde — APPROVED** [review]
LGTM

File: hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java (hunk 1 above)
Comment (hemantk-12) [reply hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdiff/TestRocksDBCheckpointDiffer.java]: No, it doesn't have to be. Updated it.

**Comment by smengcl** [comment]
Thanks @hemantk-12 for the patch. Thanks @prashantpogde for the review.

**Pull request merged**